### PR TITLE
Raf waitforconfirmation maintenance

### PR DESCRIPTION
**Pull request by rfustino**

### CODE REVIEW

File: docs/sdks/javascript/index.md [hunk 1]
```diff
@@ -311,7 +275,7 @@ async function firstTransaction() {
         let amount = 1000000;
         let closeout = receiver; //closeRemainderTo
         let sender = myAccount.addr;
-        let txn = algosdk.makePaymentTxnWithSuggestedParams(sender, receiver, amount, undefined, note, params);
+        let txn = algosdk.makePaymentTxnWithSuggestedParams(sender, receiver, amount, closeout, note, params);
```
Comment (barnjamin) [inline docs/sdks/javascript/index.md:278]: same as above, keep undefined

File: examples/start_building/v2/go/yourFirstTransaction.go [hunk 2]
```diff
@@ -111,14 +70,16 @@ func main() {
 	}
 	fromAddr := myAddress
 	toAddr := "GD64YIY3TWGDMCNPP553DZPPR6LDUSFQOIJVFDPPXWEG3FVOJCCDBBHU5A"
-	var amount uint64 = 9000000
+	// close to dispenser
+	closeToAddr := "HZ57J3K46JIJXILONBBZOHX6BKPXEM2VVXNRFSUED6DKFD5ZD24PMJ3MVA"
```
Comment (barnjamin) [inline examples/start_building/v2/go/yourFirstTransaction.go:74]: keep as nil or empty string

File: examples/start_building/v2/java/YourFirstTransaction.java [hunk 3]
```diff
@@ -93,54 +87,83 @@ public void gettingStartedExample() throws Exception {
...
+            System.out.println("Algorand suggested parameters: " + jsonObj.toString(2));
             Transaction txn = Transaction.PaymentTransactionBuilder()
-                .sender(myAddress)
+                .sender(myAccount.getAddress().toString())
                 .note(note.getBytes())
-                .amount(100000)
+                .amount(1000000) // 1 algo = 1000000 microalgos
                 .receiver(new Address(RECEIVER))
                 .suggestedParams(params)
+                .closeRemainderTo(RECEIVER)
```
Comment (barnjamin) [inline examples/start_building/v2/java/YourFirstTransaction.java:98]: rm plz

File: examples/start_building/v2/javascript/yourFirstTransaction.js [hunk 4]
```diff
@@ -1,56 +1,33 @@
...
+
         // Generate a public/private key pair
-        const passphrase = "price clap dilemma swim genius fame lucky crack torch hunt maid palace ladder unlock symptom rubber scale load acoustic drop oval cabbage review abstract embark";
-        let myAccount = algosdk.mnemonicToSecretKey(passphrase);
-        console.log("My address: %s", myAccount.addr);
-        console.log("My passphrase: " + passphrase);
+        // const passphrase = "price clap dilemma swim genius fame lucky crack torch hunt maid palace ladder unlock symptom rubber scale load acoustic drop oval cabbage review abstract embark";
+        // let myAccount = algosdk.mnemonicToSecretKey(passphrase);
+        // console.log("My address: %s", myAccount.addr);
+        // console.log("My passphrase: " + passphrase);
```
Comment (barnjamin) [inline examples/start_building/v2/javascript/yourFirstTransaction.js:30]: just remove if we don't want them here?

File: examples/start_building/v2/javascript/yourFirstTransaction.js [hunk 5]
```diff
@@ -74,11 +51,13 @@ async function yourFirstTransaction() {
         params.flatFee = true;
 
         // receiver defined as TestNet faucet address 
-        const receiver = "GD64YIY3TWGDMCNPP553DZPPR6LDUSFQOIJVFDPPXWEG3FVOJCCDBBHU5A";
+        const receiver = "HZ57J3K46JIJXILONBBZOHX6BKPXEM2VVXNRFSUED6DKFD5ZD24PMJ3MVA";
+        const closeout = "HZ57J3K46JIJXILONBBZOHX6BKPXEM2VVXNRFSUED6DKFD5ZD24PMJ3MVA";
```
Comment (barnjamin) [inline examples/start_building/v2/javascript/yourFirstTransaction.js:55]: undefined

**Review by barnjamin — CHANGES_REQUESTED** [review]
Lets keep the close to as unset across all the examples unless we're specifically demonstrating how to close an account

**Review by barnjamin — APPROVED** [review]

File: docs/sdks/go/index.md [hunk 6]
```diff
@@ -173,6 +171,9 @@ if err != nil {
 }
 fromAddr := myAddress
 toAddr := "GD64YIY3TWGDMCNPP553DZPPR6LDUSFQOIJVFDPPXWEG3FVOJCCDBBHU5A"
+
+closeToAddr := "HZ57J3K46JIJXILONBBZOHX6BKPXEM2VVXNRFSUED6DKFD5ZD24PMJ3MVA"
```
Comment (barnjamin) [inline docs/sdks/go/index.md:175]: missed this one?

File: examples/start_building/v2/go/yourFirstTransaction.go [hunk 7]
```diff
@@ -3,60 +3,18 @@ package main
...
-	"errors"
+	// "errors"
 	"fmt"
-	"strings"
-
+	// "strings"
+	"github.com/algorand/go-algorand-sdk/future"
 	"github.com/algorand/go-algorand-sdk/client/v2/algod"
-	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+	// "github.com/algorand/go-algorand-sdk/client/v2/common/models"
```
Comment (barnjamin) [inline examples/start_building/v2/go/yourFirstTransaction.go:11]: can you just remove these commented out lines please?

File: examples/start_building/v2/go/yourFirstTransaction.go [hunk 8]
```diff
@@ -111,7 +69,8 @@ func main() {
 	}
 	fromAddr := myAddress
 	toAddr := "GD64YIY3TWGDMCNPP553DZPPR6LDUSFQOIJVFDPPXWEG3FVOJCCDBBHU5A"
-	var amount uint64 = 9000000
+	// close to dispenser
```
Comment (barnjamin) [inline examples/start_building/v2/go/yourFirstTransaction.go:72]: we dont do this anymore, can you please remove

File: docs/sdks/javascript/index.md (hunk 1 above)
Comment (rfustino) [reply docs/sdks/javascript/index.md]: fixed

File: examples/start_building/v2/go/yourFirstTransaction.go (hunk 2 above)
Comment (rfustino) [reply examples/start_building/v2/go/yourFirstTransaction.go]: fixed

File: examples/start_building/v2/java/YourFirstTransaction.java (hunk 3 above)
Comment (rfustino) [reply examples/start_building/v2/java/YourFirstTransaction.java]: fixed

File: docs/sdks/javascript/index.md [hunk 9]
```diff
@@ -156,7 +156,9 @@ To interact with the Algorand blockchain, you can send different types of transa
         const note = enc.encode("Hello World");
         let amount = 1000000; // equals 1 ALGO
         let sender = myAccount.addr;
-        let txn = algosdk.makePaymentTxnWithSuggestedParams(sender, receiver, amount, undefined, note, params);
+        let closeout = receiver; //closeRemainderTo - return remainder to TestNet faucet
```
[the remark this answers is not in the document]
Comment (rfustino) [reply docs/sdks/javascript/index.md]: fixed

File: examples/start_building/v2/javascript/yourFirstTransaction.js (hunk 5 above)
Comment (rfustino) [reply examples/start_building/v2/javascript/yourFirstTransaction.js]: fixed

File: docs/sdks/go/index.md (hunk 6 above)
Comment (rfustino) [reply docs/sdks/go/index.md]: Ya, i had deleted it, then the merge commit pt it back. all set now.

File: examples/start_building/v2/go/yourFirstTransaction.go (hunk 8 above)
Comment (rfustino) [reply examples/start_building/v2/go/yourFirstTransaction.go]: ya, caught that too - removed the comment

File: examples/start_building/v2/java/YourFirstTransaction.java [hunk 10]
```diff
@@ -93,54 +87,80 @@ public void gettingStartedExample() throws Exception {
...
+
+
+// System.out.println("Navigate to this link:  https://dispenser.testnet.aws.algodev.network/?account=" + myAccount1.getAddress().toString());      
+
+
+//   .closeRemainderTo(RECEIVER) // WARNING! all remaining funds in the sender account will be sent to the closeRemainderTo Account, omit RECEIVER account when in use otherwise all funds from the sender account will be sent to that account.
+//   .build();
+// CloseRemainder can be used to reset sender account to 0.
+// Normally this would be omitted. For more info see: 
+// https://developer.algorand.org/docs/reference/transactions/#payment-transaction
```
[the remark this answers is not in the document]
Comment (rfustino) [reply examples/start_building/v2/java/YourFirstTransaction.java]: ya, caught that too. thx

File: examples/start_building/v2/go/yourFirstTransaction.go (hunk 7 above)
Comment (rfustino) [reply examples/start_building/v2/go/yourFirstTransaction.go]: done

**Pull request merged**